src/aabb.rs

```diff
@@ -141,6 +139,24 @@ pub trait Aabb<
         let max = P::from_vec(self.max().to_vec().mul_element_wise(v));
         Aabb::new(min, max)
     }
+
+    /// Give a list of the corner vertices of this bounding box.
+    fn corners(&self) -> Vec<P>;
+
+    /// Apply an arbitrary transform to the corners of this bounding box,
+    /// return a new conservative bound.
+    fn transform<T>(&self, transform: &T) -> Self
+    where
+        T: Transform<P>,
+    {
+        let base = Self::new(
+            transform.transform_point(self.min()),
+            transform.transform_point(self.max()),
+        );
+        self.corners().iter().fold(base, |u, &corner| {
```

Comment:
The old code did it a bit differently, iterating only on corner[1] and [2], given that [0] and [3] are already a part of ht base. However, on a second thought this would fail for some transformations, given that the transformed min() may no longer be the min() in the transformed space. Therefore, it might be reasonable to have the base formed from corner[0] only (with size of 0), and then grown by the transformed corners[1..].